omod/src/main/java/org/openmrs/module/coreapps/web/resource/LatestObsResource.java

```diff
@@ -85,9 +86,12 @@ protected PageableResult doSearch(RequestContext context) {
 				List<Concept> questions = new Vector<Concept>();
 				questions.add(concept);
 				
-				List<Obs> obs = obsService.getObservations(who, null, questions, null, null, null, sort, 1, null, null, null, false);
-				if (obs.size() > 0) {
+				int nLatestObsPerGroup = StringUtils.isNotEmpty(context.getParameter("nLatestObsPerGroup")) ? Integer.parseInt(context.getParameter("nLatestObsPerGroup")) : 1;
```

Comment:
Maybe just call these "nLatestObs"?  I think "perGroup" is a little confusing because there's only one "group" here, the group of all the obs associated wtih the list of question concepts, correct?  I also think using "group" is confusing because an "ObsGroup" has a specific meaning and I don't think you are using it in that context if I'm understanding correct?

Will add some more comments on the  ticket.

Reply:
Oh yes, thanks @mogoodrich. `perGroup` makes it verbose and confusing. I'll correct this in the next commit.